### PR TITLE
Handle empty array when preloading records

### LINKED ISSUE #5028: Handle empty array in DataContainer's preloadCurrentRecords
### Affected version(s)

5.0

### Description

Run into this error in the backend
```php
ErrorException:
Warning: Undefined array key 0

  at vendor/contao/core-bundle/contao/classes/DataContainer.php:1747
  at Contao\DataContainer::preloadCurrentRecords()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:4204)
  at Contao\DC_Table->generateTree()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:3682)
  at Contao\DC_Table->treeView()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:362)
  at Contao\DC_Table->showAll()
     (/home/michael/www/_repos/contao-marketing-suite/src/Resources/contao/library/MarketingSuite/BackendModule/Module.php:300)
  at numero2\MarketingSuite\BackendModule\Module->getModule()
     (/home/michael/www/_repos/contao-marketing-suite/src/Resources/contao/library/MarketingSuite/BackendModule/Module.php:92)
  at numero2\MarketingSuite\BackendModule\Module->generate()
     (vendor/contao/core-bundle/contao/classes/Backend.php:316)
  at Contao\Backend->getBackendModule()
     (vendor/contao/core-bundle/contao/controllers/BackendMain.php:149)
  at Contao\BackendMain->run()
     (/home/michael/www/_repos/contao-marketing-suite/src/Resources/contao/library/MarketingSuite/Controllers/Main.php:32)
  at numero2\MarketingSuite\Controller\Main->run()
     (/home/michael/www/_repos/contao-marketing-suite/src/Controller/BackendController.php:46)
  at numero2\MarketingSuiteBundle\Controller\BackendController->cmsMain()
     (vendor/symfony/http-kernel/HttpKernel.php:152)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw()
     (vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle()
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle()
     (public/index.php:44)    
```

The `DataContainer`'s `preloadCurrentRecords` function does not correctly handle if it recieves an empty array.
Reason:
https://github.com/contao/contao/blob/563ab9d601ae69777b9fb447403f246bcd8fef13/core-bundle/contao/classes/DataContainer.php#L1747

### PR DESCRIPTION
**Pull request by aschempp**
Fixes #5028

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

File: core-bundle/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -1732,6 +1732,11 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 	 */
 	protected static function preloadCurrentRecords(array $ids, string $table): void
 	{
+		if (empty($ids))
```
Comment (ausi) [inline core-bundle/contao/classes/DataContainer.php]:
```suggestion
		if (!$ids)
```
`$ids` is always defined here, so there is no need to use the error-suppressing `empty()` I think.

Comment (aschempp) [reply core-bundle/contao/classes/DataContainer.php]: I find that much more readable 🤷

Comment (leofeyer) [reply core-bundle/contao/classes/DataContainer.php]: I think we have agreed on using `!\count($ids)` in this case, haven‘t we?

Comment (ausi) [reply core-bundle/contao/classes/DataContainer.php]: I always forget 😞
One day I will write a CS fixer for that ☺️

**Review by leofeyer — APPROVED** [review]

**Pull request merged**